### PR TITLE
Refactor `Variable` class and rename it to `DartObjectNode`

### PR DESCRIPTION
**Pull request by elliette**
Addresses two TODOs on the `Variable` class:

1. Consider a new class name. This class is more just the data model for a tree of Dart objects with properties rather than a "Variable".
2. Refactor to make `name`, `ref,` `text` named (and potentially required?) parameters. Give `ref` a type.

### CODE REVIEW

File: packages/devtools_app/lib/src/console_service.dart [hunk 1]
```diff
@@ -24,7 +24,7 @@ class ConsoleLine {
       );
 
   factory ConsoleLine.variable(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/console_service.dart:26]: rename this factory constructor too

File: packages/devtools_app/lib/src/debugger/debugger_model.dart [hunk 2]
```diff
@@ -328,12 +328,13 @@ Future<void> addExpandableChildren(
 }
 
 /// Builds the tree representation for a [Variable] object by querying data,
-/// creating child Variable objects, and assigning parent-child relationships.
+/// creating child DartObjectNode objects, and assigning parent-child
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/debugger_model.dart:331]: wrap DartObjectNode in [ ]

File: packages/devtools_app/lib/src/debugger/debugger_model.dart [hunk 3]
```diff
@@ -695,13 +696,14 @@ List<Variable> _createVariablesForFields(
...
-// TODO(elliette): Refactor to make name, ref, text named (and potentially
-// required?) parameters. Give ref a type.
-class Variable extends TreeNode<Variable> {
-  Variable._(this.name, ref, this.text, {int offset, int childCount})
-      : _ref = ref,
+class DartObjectNode extends TreeNode<DartObjectNode> {
+  DartObjectNode._({
+    this.name,
+    this.text,
+    GenericInstanceRef ref,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/debugger/debugger_model.dart:703]: make these required

**Review by kenzieschmoll — APPROVED** [review]
LGTM with a couple comments

**Review by kenzieschmoll — APPROVED** [review]
LGTM with a couple comments

File: packages/devtools_app/lib/src/console_service.dart (hunk 1 above)
Comment (elliette) [reply packages/devtools_app/lib/src/console_service.dart]: Done

File: packages/devtools_app/lib/src/debugger/debugger_model.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/lib/src/debugger/debugger_model.dart]: Done

File: packages/devtools_app/lib/src/debugger/debugger_model.dart (hunk 3 above)
Comment (elliette) [reply packages/devtools_app/lib/src/debugger/debugger_model.dart:703]: Looks like none of them can be required (for example, `DartObjectNode.text` only provides the `text` parameter).

**Pull request merged**